source/extensions/common/wasm/context.cc

```diff
@@ -1836,14 +1836,17 @@ void Context::onHttpCallFailure(uint32_t token, Http::AsyncClient::FailureReason
 
 void Context::onGrpcReceiveWrapper(uint32_t token, ::Envoy::Buffer::InstancePtr response) {
   ASSERT(proxy_wasm::current_context_ == nullptr); // Non-reentrant.
+  if (wasm()->isGrpcCallId(token)) {
+    grpc_call_request_.erase(token);
```

Comment:
@PiotrSikora If we remove grpc_call_request object here, I think the wasm callback corresponding to `on_grpc_receive_` may crash if it want to access the request object. Is it OK?

Reply:
`grpc_call_request_` is only tracked on Envoy side, and it's not used in Proxy-Wasm C++ Host and Wasm plugin.

However, I think we're removing it prematurely (triggering destructor) by calling `erase` here. Should we instead `extract` it, so that it's freed after the function is done?

Reply:
Updated. We can defer that, too.